### ISSUE TITLE
"An unknown error occurred" 

### ISSUE THREAD

**Issue by Gerrof** [samhh/bukubrow-webext#159]
Hello, 

i'm stucked in the installation of this (awesome looking, thank you for this) browser extension. I'm using ArchLinux and tried to install the extension for LibreWolf, Firefox and chromium. 

After i installed the host and the extension via the browsers store, the extension gui shows up "An unknown error occurred". 

I checked if the native messaging host json is at the right place for each browser and i dont know where too look further. 

```bukubrow -l```and ```bukubrow -o 1```worked as expected.

**Comment by samhh** [comment]
Is the host prebuilt? If not, is it built from master or at a tagged version?

What's the content of your manifest for Firefox? It should be at: `~/.mozilla/native-messaging-hosts/com.samhh.bukubrow.json`

**Comment by Gerrof** [comment]
Thank you very much for your answer. I used the prebuilt from the [AUR](https://aur.archlinux.org/packages/bukubrow/), installed it with ```makepkg -si``` and executed ```bukubrow --install-firefox```. This created the following file:
```
[maik@archlinuxusb ~]$ bukubrow --install-firefox
Successfully installed host for Firefox to:
	"/home/maik/.mozilla/native-messaging-hosts/com.samhh.bukubrow.json"
[maik@archlinuxusb ~]$ cat .mozilla/native-messaging-hosts/com.samhh.bukubrow.json 
{"name":"com.samhh.bukubrow","description":"Bukubrow host for the Firefox extension","path":"/usr/bin/bukubrow","type":"stdio","allowed_extensions":["bukubrow@samhh.com"]}
```
For now, i think for the native host system is everything fine. If i start firefox from terminal, it is working like excepted. But if i start it with dmenu over my shortcut file "firefox.desktop" the "An unknown error occurred"-message appears, for each browser. 

At this moment i try to install all the stuff global, maybe the browser cannot invoke buku or bukubrow over stdin. Do you have any idea to this behavior ?

**Comment by samhh** [comment]
Curious. Could you paste your `firefox.desktop` file and how you're invoking it via dmenu? We're on very similar systems so hopefully I can repro.

**Comment by Gerrof** [comment]
Ok, here we go, but without the translation parts for better readability:

## /usr/share/applications/firefox.desktop

```
[Desktop Entry]
Version=1.0
Name=Firefox
GenericName=Web Browser
...
Comment=Browse the World Wide Web
...
Keywords=Internet;WWW;Browser;Web;Explorer
...
Exec=/usr/lib/firefox/firefox %u
Icon=firefox
Terminal=false
X-MultipleArgs=false
Type=Application
MimeType=text/html;text/xml;application/xhtml+xml;x-scheme-handler/http;x-scheme-handler/https;application/x-xpinstall;application/pdf;application/json;
StartupNotify=true
StartupWMClass=firefox
Categories=Network;WebBrowser;
Actions=new-window;new-private-window;

[Desktop Action new-window]
Name=New Window
...
Exec=/usr/lib/firefox/firefox --new-window %u

[Desktop Action new-private-window]
Name=New Private Window
...
Exec=/usr/lib/firefox/firefox --private-window %u
```

## Using i3-dmenu-desktop with shortcut [win cmd] + d
### ~/.config/i3/config

```
...
bindcode $mod+40 exec --no-startup-id i3-dmenu-desktop
...
```

I will upload the whole files later, If you need these to reproduce.

**Comment by Gerrof** [comment]
I found the cause: It is related to the shell enviroment, like described in this [stackexchange post](https://unix.stackexchange.com/questions/390501/differences-in-launching-a-program-through-terminal-and-through-an-application-l) 

I set $XDG_DATA_HOME in my .bashrc, so buku use that variable and all started browsers from the current bash shell. But if started via dmenu, $XDG_DATA_HOME will never set thru .bashrc because there wasn't an interactive shell invoked. Now i set this variable with .profile, so LXDM will source my variables and everything works fine. I will look for a better solution for my cross-distro-configuration.

Im sorry for my misconfiguration and thank you very much for your fast help.
